### PR TITLE
(DOCSP-21746) Add Scala examples to Structured Streaming page

### PR DESCRIPTION
**Pull request by zach-carr**
## Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-spark-connector/blob/master/REVIEWING.md)

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-21746

### Snooty build log:
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=624f37b20fdfea9c3742a5ea

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/spark-connector/docsworker-xlarge/DOCSP-21746/structured-streaming/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging and workerpool job links in the PR description updated?

### If your page documents a concept, does it meet the following criteria?

- [ ] Target the [Jasmin persona](https://drive.google.com/file/d/14FbBOLCVxwSP6M9BK4Nz1Ir9tzxT8_02/view)
- [ ] Target the [Lucas persona](https://drive.google.com/file/d/1J2vqJxo7ldv7OP_obA9Q-avf0o_ju4Lk/view)

### CODE REVIEW

**Comment by zach-carr** [comment]
Hey @rozza, the Scala examples are ready for a look whenever you get a chance. Thanks!

File: source/structured-streaming.txt [hunk 1]
```diff
@@ -175,6 +228,72 @@ more about continuous processing, see the `Spark documentation <https://spark.ap
...
+         The following code snippet shows how to use the preceding 
+         configuration settings to stream data from MongoDB:
+
+         .. code-block:: scala
+            :copyable: true
+            :emphasize-lines: 1, 4, 8
+
+            import org.apache.spark.sql.streaming.Trigger
+         
+            streamingDataFrame = <local SparkSession>.readStream
```
Comment (rozza) [inline source/structured-streaming.txt:278]: Update `streamingDataFrame =` to `val streamingDataFrame =`

File: source/structured-streaming.txt [hunk 2]
```diff
@@ -175,6 +228,72 @@ more about continuous processing, see the `Spark documentation <https://spark.ap
...
+              .format("mongodb")
+              .load()
+         
+            query = streamingDataFrame.writeStream
```
Comment (rozza) [inline source/structured-streaming.txt:282]: `val query =`

File: source/structured-streaming.txt [hunk 3]
```diff
@@ -227,14 +346,59 @@ Stream to MongoDB from a CSV File
...
+         As streaming data is read from the ``.csv`` file, it is added 
+         to MongoDB in the `outputMode <https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/streaming/DataStreamWriter.html#outputMode(outputMode:String):org.apache.spark.sql.streaming.DataStreamWriter[T]>`__ 
+         you specify.
+
+         .. code-block:: scala
+            :copyable: true
+            :emphasize-lines: 10, 16
+
+            // create a local SparkSession
+            spark = SparkSession
```
Comment (rozza) [inline source/structured-streaming.txt:376]: `val spark = `

File: source/structured-streaming.txt [hunk 4]
```diff
@@ -227,14 +346,59 @@ Stream to MongoDB from a CSV File
...
+              .builder
+              .appName("writeExample")
+              .master("spark://spark-master:<port>")
+              .config("spark.jars", "<mongo-spark-connector>.jar")
+              .getOrCreate()
+
+            // define a streaming query
+            query = spark.readStream
```
Comment (rozza) [inline source/structured-streaming.txt:384]: `val query = `

File: source/structured-streaming.txt [hunk 5]
```diff
@@ -227,14 +346,59 @@ Stream to MongoDB from a CSV File
...
+              .format("csv")
+              .option("header", "true")
+              .schema(<csv-schema>)
+              .load(<csv-file-name>)
+              # manipulate your streaming data
```
Comment (rozza) [inline source/structured-streaming.txt:389]: Change to `// manipulate your streaming data`

File: source/structured-streaming.txt [hunk 6]
```diff
@@ -294,4 +458,55 @@ Stream to a CSV File from MongoDB
...
+         As new data is inserted into MongoDB, MongoDB streams that 
+         data out to a ``.csv`` file in the `outputMode <https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/streaming/DataStreamWriter.html#outputMode(outputMode:String):org.apache.spark.sql.streaming.DataStreamWriter[T]>`__ 
+         you specify.
+
+         .. code-block:: scala
+            :copyable: true
+            :emphasize-lines: 17, 25, 28
+
+            // create a local SparkSession
+            spark = SparkSession
```
Comment (rozza) [inline source/structured-streaming.txt:482]: `val spark`

File: source/structured-streaming.txt [hunk 7]
```diff
@@ -294,4 +458,55 @@ Stream to a CSV File from MongoDB
...
+              .builder
+              .appName("readExample")
+              .master("spark://spark-master:<port>")
+              .config("spark.jars", "<mongo-spark-connector>.jar")
+              .getOrCreate()
+
+            // define the schema of the source collection
+            readSchema = StructType()
```
Comment (rozza) [inline source/structured-streaming.txt:490]: `val readSchema = `

File: source/structured-streaming.txt [hunk 8]
```diff
@@ -294,4 +458,55 @@ Stream to a CSV File from MongoDB
...
+              .add("company_symbol", StringType())
+              .add("company_name", StringType())
+              .add("price", DoubleType())
+              .add("tx_time", TimestampType())
+
+            // define a streaming query
+            query = spark.readStream
```
Comment (rozza) [inline source/structured-streaming.txt:497]: `val query = `

File: source/structured-streaming.txt [hunk 9]
```diff
@@ -294,4 +458,55 @@ Stream to a CSV File from MongoDB
...
+              .format("mongodb")
+              .option("spark.mongodb.connection.uri", <mongodb-connection-string>)
+              .option("spark.mongodb.database", <database-name>)
+              .option("spark.mongodb.collection", <collection-name>)
+              .schema(readSchema)
+              .load()
+              # manipulate your streaming data
```
Comment (rozza) [inline source/structured-streaming.txt:504]: `// manipulate your streaming data`

**Review by rozza — CHANGES_REQUESTED** [review]
Just some minor nits

**Comment by zach-carr** [comment]
Ah my bad 🤦 I had those changes added after testing, just not committed and pushed (the `val`s anyway, didn't catch the comments)

Thanks!

**Review by rozza — APPROVED** [review]
LGTM!

**Review by terakilobyte — APPROVED** [review]
lgtm

**Pull request merged**